python/ray/autoscaler/updater.py

```diff
@@ -223,17 +223,10 @@ def set_ssh_ip_if_required(self):
         #   the ControlPath directory exists, allowing SSH to maintain
         #   persistent sessions later on.
         try:
-            self.process_runner.check_call(
-                ["mkdir", "-p", self.ssh_control_path])
-        except subprocess.CalledProcessError as e:
+            os.makedirs(self.ssh_control_path, mode=0o700, exist_ok=True)
+        except Exception as e:
```

Comment:
Does it make sense to be a bit more specific with the exception here? It is usually a bad idea to catch `Exception` generically! (E.g. it would mask typos in the line above)

Reply:
Yeah, very fair! I just couldn't find any documentation to tell me what exceptions `os.makedirs` could throw so we could narrow it down to just specific problems.

I debated dropping the try catch so that errors could bubble up, but I'm not sure we actually want this to be anything other than an "warn & continue" behavior.

Suggestions very welcome :D

Reply:
Moved to using OSError, as that'll hopefully catch what we care about and not catch what we don't care about!